docs/rfcs/autoops-flag-trigger.md

```diff
@@ -0,0 +1,228 @@
+# Summary
+
+We will implement flag triggers feature that enables users integrate Bucketeer with their system or tools. Basically, it
+will let user turn on/off the feature flag based on user's system or tools status.
+
+## Implementation
+
+### Infra
+
+* Add a new table `flag_triggers` to store the flag triggers data.
+
+```sql
+CREATE TABLE `flag_triggers`
+(
+    `id`                    varchar(255) NOT NULL,
+    `feature_id`            varchar(255) NOT NULL,
+    `environment_namespace` varchar(255) NOT NULL,
+    `name`                  VARCHAR(255) NOT NULL default '',
+    `type`                  int          NOT NULL,
+    `action`                tinyint(1) NOT NULL,
+    `description`           text         NOT NULL default '',
+    `trigger_times`         int          NOT NULL,
+    `last_triggered_at`     bigint       NOT NULL,
+    `uuid`                  varchar(512) NOT NULL default '',
+    `disabled`              tinyint(1) NOT NULL DEFAULT '0',
+    `deleted`               tinyint(1) NOT NULL DEFAULT '0',
+    `created_at`            bigint       NOT NULL,
+    `updated_at`            bigint       NOT NULL,
+    PRIMARY KEY (`id`)
+)ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;
```

Comment:
For consistency with other tables, please can you add the Primary Key and Foreign Key like:
```
  PRIMARY KEY (`id`,`environment_namespace`),
  CONSTRAINT `foreign_flag_triggers_feature_id_environment_namespace`
  FOREIGN KEY (`feature_id`, `environment_namespace`)
  REFERENCES `feature` (`id`, `environment_namespace`)
  ON DELETE NO ACTION
  ON UPDATE NO ACTION
```

Reply:
Thanks for your supplement!